apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.ExecutableCommand.cs

```diff
@@ -106,12 +106,20 @@ internal async Task<SpannerDataReader> ExecuteReaderAsync(CommandBehavior behavi
                 ValidateConnectionAndCommandTextBuilder();
                 ValidateCommandBehavior(behavior);
 
-                if (CommandTextBuilder.SpannerCommandType != SpannerCommandType.Select && CommandTextBuilder.SpannerCommandType != SpannerCommandType.Read)
+                await Connection.EnsureIsOpenAsync(cancellationToken).ConfigureAwait(false);
+                var enableGetSchemaTable = Connection.Builder.EnableGetSchemaTable;
+                // When the data reader is closed, we may need to dispose of the connection.
+                IDisposable resourceToClose = (behavior & CommandBehavior.CloseConnection) == CommandBehavior.CloseConnection ? Connection : null;
```

Comment:
You don't need to do any of this before checking for DML.
The ExecuteDmlReadyAsync should receive a CommandBehaviour and check there.

I would move the if whith the DML check right after ValidateConnectionAndCommandTextBuilder(), and then leave the rest of the method as it was before. You can then do the DML Reader specific checks in `ExecuteDmlReaderAsync(CommandBehaviour, CancellationToken)`.